openscenario/openscenario_interpreter/src/syntax/condition.cpp

```diff
@@ -43,10 +47,24 @@ Condition::Condition(const pugi::xml_node & node, Scope & scope)
 
 auto Condition::evaluate() -> Object
 {
-  if (condition_edge == ConditionEdge::sticky and current_value) {
-    return true_v;
-  } else {
-    return asBoolean(current_value = Object::evaluate().as<Boolean>());
+  switch (condition_edge) {
+    case ConditionEdge::rising:
+      return update_condition(std::function([](bool a, bool b) -> bool { return a and not b; }));
+
+    case ConditionEdge::falling:
+      return update_condition(std::function([](bool a, bool b) -> bool { return not a and b; }));
+
+    case ConditionEdge::risingOrFalling:
+      return update_condition(std::function([](bool a, bool b) -> bool { return a xor b; }));
+
+    case ConditionEdge::none:
+      return update_condition(std::function([](bool a) -> bool { return a; }));
+
+    case ConditionEdge::sticky:
+      return update_condition(std::function([this](bool a) -> bool { return current_value or a; }));
```

Comment:
By using `!=` instead of `xor`, the explicit return type `bool` is unnecessary.

```suggestion
    case ConditionEdge::rising:
      return update_condition(std::function([](bool a, bool b) { return a and not b; }));

    case ConditionEdge::falling:
      return update_condition(std::function([](bool a, bool b) { return not a and b; }));

    case ConditionEdge::risingOrFalling:
      return update_condition(std::function([](bool a, bool b) { return a != b; }));

    case ConditionEdge::none:
      return update_condition(std::function([](bool a) { return a; }));

    case ConditionEdge::sticky:
      return update_condition(std::function([this](bool a) { return current_value or a; }));
```